### PR TITLE
Adapting the script classification_with_grn_and_vsn to be Backend-Agnostic

### PR DESCRIPTION
**Pull request by Humbulani1234**
This PR adapts the script `classification_with_grn_and_vsn.py`  from `structured_data` examples to `jax` and `torch`, i.e., make the script ***Backend-Agnostic***.

**Approach**:

- Modified the model architecture and removed the `Preprocessing` layers and use them with the `tf.data.Datasets`. Refer to [this](https://github.com/keras-team/keras/blob/master/keras/src/layers/preprocessing/string_lookup.py#L42) Keras doc string for more.
- Generated the `.ipynb` and `.md` files
-  Achieved the `95%` accuracy in the original model

### CODE REVIEW

File: examples/structured_data/classification_with_grn_and_vsn.py [hunk 1]
```diff
@@ -157,6 +182,20 @@
...
+
+subprocess.run(
+    f'find {extracted_path} -type f ! -name "*.tar.gz" -exec rm -f {{}} +',
+    shell=True,
+    check=True,
+)
+subprocess.run(
+    f"find {extracted_path} -type d -empty -exec rmdir {{}} +", shell=True, check=True
+)
+
```
Comment (hertschuh) [inline examples/structured_data/classification_with_grn_and_vsn.py:198]: This could apply to any dataset, but I find this an unnecessary distraction. Can you remove?

File: examples/structured_data/classification_with_grn_and_vsn.py [hunk 2]
```diff
@@ -376,17 +424,78 @@ def __init__(self, num_features, units, dropout_rate):
...
+            import torch
+
+            result_torch = matmul_dependent_on_backend(v, x)
+            return result_torch
+        # tensorflow backend
+        if keras.backend.backend() == "tensorflow":
+            import tensorflow as tf
+
+            result_tf = keras.ops.squeeze(tf.matmul(v, x, transpose_a=True), axis=1)
+            return result_tf
```
Comment (hertschuh) [inline examples/structured_data/classification_with_grn_and_vsn.py:494]: This definitely should not be needed.

What is the issue with `keras.ops.squeeze(keras.ops.matmul(keras.ops.transpose(v), x), axis=1)`?

File: examples/structured_data/classification_with_grn_and_vsn.py [hunk 3]
```diff
@@ -376,17 +424,78 @@ def __init__(self, num_features, units, dropout_rate):
...
+
+    # to remove the build warnings
+    def build(self):
+        self.built = True
```
Comment (hertschuh) [inline examples/structured_data/classification_with_grn_and_vsn.py:498]: Thanks!

File: examples/structured_data/classification_with_grn_and_vsn.py [hunk 4]
```diff
@@ -415,7 +520,7 @@ def create_model(encoding_size):
 learning_rate = 0.001
 dropout_rate = 0.15
 batch_size = 265
-num_epochs = 20
+num_epochs = 1  # maybe adjusted to a desired value
```
Comment (hertschuh) [inline examples/structured_data/classification_with_grn_and_vsn.py:523]: Please revert after you're done testing.

File: examples/structured_data/classification_with_grn_and_vsn.py [hunk 5]
```diff
@@ -108,13 +109,37 @@
     "income_level",
 ]
 
-data_url = "https://archive.ics.uci.edu/static/public/20/census+income.zip"
+data_url = "https://archive.ics.uci.edu/static/public/117/census+income+kdd.zip"
```
Comment (hertschuh) [inline examples/structured_data/classification_with_grn_and_vsn.py:112]: Why the change in dataset?

It seems like the original dataset was easier to handle.

**Review by hertschuh — CHANGES_REQUESTED** [review]
@Humbulani1234 ,

Thank you for the PR!

File: examples/structured_data/classification_with_grn_and_vsn.py (hunk 1 above)
Comment (Humbulani1234) [reply examples/structured_data/classification_with_grn_and_vsn.py:198]: Removed

File: examples/structured_data/classification_with_grn_and_vsn.py (hunk 2 above)
Comment (Humbulani1234) [reply examples/structured_data/classification_with_grn_and_vsn.py:494]: After careful thought, I've made it to work. I also struggled with the Keras doc string of the op [keras.transpose](https://github.com/keras-team/keras/blob/v3.8.0/keras/src/ops/numpy.py#L6302#6303), I don't think axes is explicit about the permutations. I had to read tensorflow [doc](https://www.tensorflow.org/api_docs/python/tf/transpose) to have a clear picture. But, nonetheless, it is resolved.

File: examples/structured_data/classification_with_grn_and_vsn.py (hunk 3 above)
Comment (Humbulani1234) [reply examples/structured_data/classification_with_grn_and_vsn.py:498]: Added more of this.

File: examples/structured_data/classification_with_grn_and_vsn.py (hunk 4 above)
Comment (Humbulani1234) [reply examples/structured_data/classification_with_grn_and_vsn.py:523]: Reverted to 20, but left the comment.

File: examples/structured_data/classification_with_grn_and_vsn.py (hunk 5 above)
Comment (Humbulani1234) [reply examples/structured_data/classification_with_grn_and_vsn.py:112]: The dataset description in the script says it must have `41 input features: 7 numerical features
and 34 categorical features.`. The original dataset only had `14 features` and its target variable was in `<= or >=50k`, whereas in the script it is in `-5000 or +5000`

**Comment by Humbulani1234** [comment]
PR sent addressing the comments.

File: examples/structured_data/classification_with_grn_and_vsn.py [hunk 6]
```diff
@@ -415,7 +471,7 @@ def create_model(encoding_size):
 learning_rate = 0.001
 dropout_rate = 0.15
 batch_size = 265
-num_epochs = 20
+num_epochs = 1  # may be adjusted to a desired value
```
Comment (hertschuh) [inline examples/structured_data/classification_with_grn_and_vsn.py:474]: Was this a different one or was it not actually reverted? (Please change back to 20)

**Review by hertschuh — CHANGES_REQUESTED** [review]

File: examples/structured_data/classification_with_grn_and_vsn.py (hunk 6 above)
Comment (Humbulani1234) [reply examples/structured_data/classification_with_grn_and_vsn.py]: Changed. It was an over-sight.

**Comment by Humbulani1234** [comment]
PR sent. Together with the `.md` and `.ipynb` files.

**Review by hertschuh — APPROVED** [review]
LGTM. Thank you for the port!
